[AMD] Add TP=8 models to nightly test and make TP=2 test stable

quitenode opened the pull request:
---

<!-- Thank you for your contribution! Please follow these guidelines to enhance your pull request. If anything is unclear, submit your PR and reach out to maintainers for assistance. Join our Slack community at https://slack.sglang.io to discuss further. -->

## Motivation

<!-- Describe the purpose and goals of this pull request. -->

1. Nightly Test (AMD) tp=2
Based on previously 2 week run data, changed thresholds to make following models stable to pass nightly CI.

mistralai/Mixtral-8x7B-Instruct-v0.1: 0.61
neuralmagic/Meta-Llama-3.1-8B-Instruct-FP8: 0.8
neuralmagic/Mixtral-8x7B-Instruct-v0.1-FP8: 0.62

2. Added and tested 8 nightly test with 7 new models with tp=8 configuration from in-house CI 

All test passed: https://github.com/sgl-project/sglang/actions/runs/20377066465?pr=15296

Please help to review. @HaiShaw @saienduri @yctseng0211 @bingxche 

### New TP=8 Models Added (GSM8K Completion Benchmark)

| Model | Runner | Config | Verified Accuracy |
|-------|--------|--------|-------------------|
| `lmsys/gpt-oss-20b-bf16` | `nightly-test-8-gpu-gpt-oss` | Triton backend | 0.550 |
| `lmsys/gpt-oss-120b-bf16` | `nightly-test-8-gpu-gpt-oss` | Triton backend | 0.840 |
| `lmzheng/grok-1` | `nightly-test-8-gpu-grok` | FP8, Aiter backend | 0.860 |
| `amd/grok-1-W4A8KV8` | `nightly-test-8-gpu-grok` | INT4, Aiter backend | 0.820 |
| `xai-org/grok-2` | `nightly-test-8-gpu-grok` | FP8, Aiter backend | 0.945 |
| `deepseek-ai/DeepSeek-V3-0324` | `nightly-test-8-gpu-deepseek-v3-dp` | DP Attention (dp-size=8) | 0.94 |
| `deepseek-ai/DeepSeek-V3-0324` | `nightly-test-8-gpu-deepseek-v3-tc` | Torch Compile | 0.96 |
| `deepseek-ai/DeepSeek-R1-0528` | `nightly-test-8-gpu-deepseek-r1` | Reasoning model | 0.965 |


### New Test File

- `test/srt/nightly/test_gsm8k_completion_eval_amd.py` - GSM8K completion benchmark for base models (TP=8)

### Test Improvements to Existing TP=2 Tests

- `test/srt/nightly/test_gsm8k_eval_amd.py` - Added runtime tracking and improved summary format

### Key Features

- Model groups selectable via `AMD_TEST_MODEL_GROUP` env var
- Runtime tracking (startup/bench/total) for all models
- Enhanced GitHub summary with failure details and statistics
- HuggingFace cache/access checking with detailed logging

---

## Modifications

<!-- Detail the changes made in this pull request. -->

## Accuracy Tests

<!-- If this pull request affects model outputs (e.g., changes to the kernel or model forward code), provide accuracy test results. -->

https://github.com/sgl-project/sglang/actions/runs/20377066465?pr=15296

### Model Group: gpt-oss

| Model | TP | Accuracy | Threshold | Startup | Bench | Total | Status |
| ----- | -- | -------- | --------- | ------- | ----- | ----- | ------ |
| lmsys/gpt-oss-20b-bf16 | 8 | 0.550 | 0.49 | 121s | 14s | 135s | ✅ PASS |
| lmsys/gpt-oss-120b-bf16 | 8 | 0.840 | 0.82 | 231s | 19s | 251s | ✅ PASS |


**Statistics:** ✅ Passed: 2 | ❌ Failed: 0 | ⏭️ Skipped: 0

**Total Runtime:** 385.9s (6.4 min)

### Model Group: grok

| Model | TP | Accuracy | Threshold | Startup | Bench | Total | Status |
| ----- | -- | -------- | --------- | ------- | ----- | ----- | ------ |
| lmzheng/grok-1 | 8 | 0.835 | 0.8 | 1122s | 19s | 1142s | ✅ PASS |
| amd/grok-1-W4A8KV8 | 8 | 0.820 | 0.8 | 662s | 15s | 677s | ✅ PASS |
| xai-org/grok-2 | 8 | 0.930 | 0.915 | 582s | 20s | 602s | ✅ PASS |


**Statistics:** ✅ Passed: 3 | ❌ Failed: 0 | ⏭️ Skipped: 0

**Total Runtime:** 2420.9s (40.3 min)


### Model Group: deepseek-r1

| Model | TP | Accuracy | Threshold | Startup | Bench | Total | Status |
| ----- | -- | -------- | --------- | ------- | ----- | ----- | ------ |
| deepseek-ai/DeepSeek-R1-0528 | 8 | 0.965 | 0.93 | 1292s | 35s | 1328s | ✅ PASS |


**Statistics:** ✅ Passed: 1 | ❌ Failed: 0 | ⏭️ Skipped: 0

**Total Runtime:** 1327.9s (22.1 min)

### Model Group: deepseek-v3-dp

| Model | TP | Accuracy | Threshold | Startup | Bench | Total | Status |
| ----- | -- | -------- | --------- | ------- | ----- | ----- | ------ |
| deepseek-ai/DeepSeek-V3-0324 | 8 | 0.940 | 0.93 | 1193s | 35s | 1228s | ✅ PASS |


**Statistics:** ✅ Passed: 1 | ❌ Failed: 0 | ⏭️ Skipped: 0

**Total Runtime:** 1228.1s (20.5 min)

### Model Group: deepseek-v3-tc

| Model | TP | Accuracy | Threshold | Startup | Bench | Total | Status |
| ----- | -- | -------- | --------- | ------- | ----- | ----- | ------ |
| deepseek-ai/DeepSeek-V3-0324 | 8 | 0.960 | 0.93 | 1464s | 52s | 1515s | ✅ PASS |


**Statistics:** ✅ Passed: 1 | ❌ Failed: 0 | ⏭️ Skipped: 0

**Total Runtime:** 1515.2s (25.3 min)

 | model | score | threshold |
| ----- | ----- | --------- |
| meta-llama/Llama-3.1-8B-Instruct | 0.856 | 0.82 |
| mistralai/Mistral-7B-Instruct-v0.3 | 0.592 | 0.58 |
| deepseek-ai/DeepSeek-Coder-V2-Lite-Instruct | 0.88 | 0.85 |
| google/gemma-2-27b-it | 0.924 | 0.91 |
| meta-llama/Llama-3.1-70B-Instruct | 0.968 | 0.95 |
| mistralai/Mixtral-8x7B-Instruct-v0.1 | 0.636 | 0.61 |
| Qwen/Qwen2-57B-A14B-Instruct | 0.884 | 0.86 |
| neuralmagic/Meta-Llama-3.1-8B-Instruct-FP8 | 0.844 | 0.8 |
| neuralmagic/Mistral-7B-Instruct-v0.3-FP8 | 0.584 | 0.54 |
| neuralmagic/Meta-Llama-3.1-70B-Instruct-FP8 | 0.964 | 0.94 |
| neuralmagic/Mixtral-8x7B-Instruct-v0.1-FP8 | 0.652 | 0.65 |
| neuralmagic/Qwen2-72B-Instruct-FP8 | 0.964 | 0.94 |
| neuralmagic/Qwen2-57B-A14B-Instruct-FP8 | 0.868 | 0.86 |
| Qwen/Qwen3-30B-A3B-Thinking-2507 | 0.964 | 0.84 |

https://github.com/sgl-project/sglang/actions/runs/20324348703?pr=15296
both TP=2 and TP=8 pass with the https://github.com/sgl-project/sglang/pull/15318 fix of segmentation fault.

## Benchmarking and Profiling

<!-- If this pull request impacts inference speed, provide benchmarking and profiling results. -->

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).
- [ ] Work with maintainers to merge your PR. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)

michaelzhang-ai committed:
Add AMD TP=8 models to nightly test (gpt-oss-120b, gpt-oss-20b)

Add lmsys/gpt-oss-120b-bf16 and lmsys/gpt-oss-20b-bf16 models with tp=8
configuration from in-house CI sanity_check.py (mi30x).

Config: --tp 8 --chunked-prefill-size 130172 --max-running-requests 128
        --mem-fraction-static 0.85 --attention-backend triton
        SGLANG_USE_AITER=0

michaelzhang-ai committed:
Merge branch 'main' into add-tp8-models-nightly

michaelzhang-ai committed:
Merge branch 'main' into add-tp8-models-nightly

michaelzhang-ai committed:
Temporarily add pull_request trigger for CI testing

Will be removed before merging.

michaelzhang-ai committed:
Remove TP=8 base models (incompatible with mgsm_en eval)

TP=8 models (gpt-oss, grok-1) are base models without chat templates.
They work with gsm8k completion API but not mgsm_en chat completions.
Kept as comments for future reference.

michaelzhang-ai committed:
Add linux-mi325-gpu-8 runner for TP=8 nightly tests

- Split nightly-test into nightly-test-2-gpu and nightly-test-8-gpu jobs
- Add workflow_dispatch job_filter for selective job runs
- 8-GPU job uses nightly-amd-8-gpu suite (needs to be defined in run_suite.py)
- Follow NVIDIA workflow pattern for multi-GPU job organization

michaelzhang-ai committed:
Merge branch 'main' into add-tp8-models-nightly

michaelzhang-ai committed:
Add nightly-amd-8-gpu suite and test file

- Add test_gsm8k_completion_eval_amd.py for base models (TP=8)
- Register nightly-amd-8-gpu suite in test/srt/run_suite.py
- Add nightly-amd-8-gpu to NIGHTLY_SUITES in test/run_suite.py

Note: GPT-OSS models are internal, test will skip if unavailable.

michaelzhang-ai committed:
Fix: pass env vars explicitly to popen_launch_server

SGLANG_USE_AITER=0 was not being passed properly to the server
subprocess. This caused GPT-OSS models to fail on CI.

michaelzhang-ai committed:
Add GPT-OSS-120B to TP=8 nightly test

- 120B model with 82% accuracy threshold
- 15 minute timeout for large model loading
- Both GPT-OSS models now tested

michaelzhang-ai committed:
Run 20B model first for faster feedback

michael-amd commented:
https://github.com/sgl-project/sglang/actions/runs/20322217744/job/58380232664?pr=15296 pass tp=8

| Model | TP | Accuracy | Threshold | Invalid | Latency | Status |

| ----- | -- | -------- | --------- | ------- | ------- | ------ |

| lmsys/gpt-oss-20b-bf16 | 8 | 0.550 | 0.5 | 0.150 | 13.8s | ✅ PASS |

| lmsys/gpt-oss-120b-bf16 | 8 | 0.865 | 0.82 | 0.015 | 19.3s | ✅ PASS |

CC: @saienduri please have a look at the PR. Thanks!

michaelzhang-ai committed:
Lower accuracy thresholds for Mixtral and Llama-3.1-8B-FP8

- Mixtral-8x7B-Instruct-v0.1: 0.64 -> 0.60 (actual: 0.61)
- Meta-Llama-3.1-8B-Instruct-FP8: 0.83 -> 0.78 (actual: 0.80)

michaelzhang-ai committed:
Add GROK1-FP8 and GROK1-IN4 to TP=8 nightly test

Test order:
1. GPT-OSS-20B (50% threshold)
2. GPT-OSS-120B (82% threshold)
3. GROK1-FP8 (80% threshold, 10min timeout)
4. GROK1-IN4 (80% threshold, 10min timeout)

GROK models use aiter backend with extended timeout for kernel compilation.

michaelzhang-ai committed:
Skip unavailable models instead of failing test

- Add check_model_available() to detect missing models
- Mark unavailable models as SKIPPED (not FAIL)
- Skipped models don't cause test failure
- Summary shows skipped count

michaelzhang-ai committed:
Fix GROK model paths to use HuggingFace format

- GROK1-FP8: lmzheng/grok-1
- GROK1-IN4: amd/grok-1-W4A8KV8
- Tokenizer: Xenova/grok-1-tokenizer

michaelzhang-ai committed:
Disable TP=8 internal models by default for upstream CI

These models (GPT-OSS, GROK) are internal/private and not available
on upstream CI runners. They are only available on in-house machines.

To enable for in-house testing, set AMD_ENABLE_TP8_MODELS=1

michaelzhang-ai committed:
Refactor TP=8 model configuration to separate internal and upstream models

- Introduced AMD_BASE_MODELS_TP8 for upstream CI, excluding internal models.
- Added comments for GROK models, indicating they are not yet cached on upstream CI.
- Removed internal model configurations from the main model list for clarity.

michaelzhang-ai committed:
Update accuracy thresholds for Mixtral and Meta-Llama models in nightly tests

- Adjusted Mixtral-8x7B-Instruct-v0.1 threshold from 0.60 to 0.61.
- Updated Meta-Llama-3.1-8B-Instruct-FP8 threshold from 0.78 to 0.80.
- Removed commented-out AMD TP=8 model configurations for clarity.

michaelzhang-ai committed:
Refactor AMD model configurations for clarity and consistency

- Removed commented-out AMD TP=2 model configurations to streamline the code.
- Updated comments to clarify that the test is specifically for the 8-GPU runner using TP=8 models.
- Ensured that all models in the test suite are consistently set to use AMD_BASE_MODELS_TP8.

michaelzhang-ai committed:
Merge branch 'main' into add-tp8-models-nightly

michaelzhang-ai committed:
Merge branch 'main' into add-tp8-models-nightly

michaelzhang-ai committed:
Merge branch 'main' into add-tp8-models-nightly

michaelzhang-ai committed:
Remove pull request trigger from nightly test workflow and update model score threshold for Mixtral in evaluation scripts.

michaelzhang-ai committed:
Update nightly test workflow to trigger on pull requests to main and modify GROK model configurations for improved download handling and caching.

michaelzhang-ai committed:
Enhance nightly test workflow by adding new GROK model configurations for 8-GPU tests, including GROK1-FP8, GROK2.5, and GROK1-IN4. Update evaluation script to support model group selection via environment variable, improving flexibility in testing different model sets.

michaelzhang-ai committed:
Refactor nightly test workflow to consolidate GROK model configurations and introduce new DeepSeek models for 8-GPU tests. Update evaluation scripts to enhance model group selection and runtime reporting, improving overall test clarity and performance metrics.

HaiShaw committed:
Merge branch 'main' into add-tp8-models-nightly

michaelzhang-ai committed:
Enhance nightly test workflow by introducing new DeepSeek models with DP attention and Torch Compile for 8-GPU tests. Update evaluation scripts to reflect model group changes and improve runtime reporting, ensuring better clarity and performance metrics.

yctseng0211 reviewed (approved):
LGTM
cc: @michael-amd @HaiShaw @bingxche

michaelzhang-ai committed:
Refactor nightly test workflow by renaming the 8-GPU job to 'nightly-test-8-gpu-gpt-oss' for clarity. Update GitHub summary generation in evaluation script to remove redundant header, improving readability of test results.

michaelzhang-ai committed:
Update nightly test script to refine model group listings and enhance clarity in environment variable descriptions. Add new DeepSeek models to the test configuration while removing outdated runtime reference data for improved readability.

HaiShaw reviewed (approved):
Changes made for amd runner only.
